libcontainer/signal_all_linux.go

```diff
@@ -0,0 +1,108 @@
+package libcontainer
+
+import (
+	"errors"
+	"os"
+	"unsafe"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/configs"
+	"github.com/opencontainers/runc/libcontainer/system"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+//nolint:structcheck,unused,revive
+type siginfo struct {
+	si_signo int32
+	si_errno int32
+	si_code  int32
+	// below here is a union; si_pid is the only field we use
+	si_pid int32
+	// Pad to 128 bytes as detailed in blockUntilWaitable
+	pad [96]byte
+}
```

Comment:
Maybe we can just rename the fields (i.e. remove the `si_` prefix) instead of suppressing linters.

Reply:
done; let's see how it goes.